src/diffusers/loaders.py

```diff
@@ -566,4 +566,453 @@ def load_textual_inversion(
+                        proxies=proxies,
+                        local_files_only=local_files_only,
+                        use_auth_token=use_auth_token,
+                        revision=revision,
+                        subfolder=subfolder,
+                        user_agent=user_agent,
+                    )
+                    state_dict = safetensors.torch.load_file(model_file, device="cpu")
+                except IOError as e:
+                    if not allow_pickle:
+                        raise e
+                    # try loading non-safetensors weights
+                    pass
+            if model_file is None:
+                model_file = _get_model_file(
+                    pretrained_model_name_or_path_or_dict,
+                    weights_name=weight_name or LORA_WEIGHT_NAME,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    resume_download=resume_download,
+                    proxies=proxies,
+                    local_files_only=local_files_only,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    subfolder=subfolder,
+                    user_agent=user_agent,
+                )
+                state_dict = torch.load(model_file, map_location="cpu")
+        else:
+            state_dict = pretrained_model_name_or_path_or_dict
+
+        # If the serialization format is new (introduced in https://github.com/huggingface/diffusers/pull/2918),
+        # then the `state_dict` keys should have `self.unet_name` and/or `self.text_encoder_name` as
+        # their prefixes.
+        keys = list(state_dict.keys())
+
+        # Load the layers corresponding to UNet.
+        if all(key.startswith(self.unet_name) for key in keys):
+            logger.info(f"Loading {self.unet_name}.")
+            unet_lora_state_dict = {k: v for k, v in state_dict.items() if k.startswith(self.unet_name)}
+            self.unet.load_attn_procs(unet_lora_state_dict)
+
+        # Load the layers corresponding to text encoder and make necessary adjustments.
+        elif all(key.startswith(self.text_encoder_name) for key in keys):
+            logger.info(f"Loading {self.text_encoder_name}.")
+            text_encoder_lora_state_dict = {
+                k: v for k, v in state_dict.items() if k.startswith(self.text_encoder_name)
+            }
+            attn_procs_text_encoder = self.load_attn_procs(text_encoder_lora_state_dict)
+            self._modify_text_encoder(attn_procs_text_encoder)
+
+        # Otherwise, we're dealing with the old format. This means the `state_dict` should only
+        # contain the module names of the `unet` as its keys WITHOUT any prefix.
+        elif not all(
+            key.startswith(self.unet_name) or key.startswith(self.text_encoder_name) for key in state_dict.keys()
+        ):
+            self.unet.load_attn_procs(state_dict)
+            logger.warning(
```

Comment:
Super nice warning, I'd maybe just use the `deprecate` function here: 
https://github.com/huggingface/diffusers/blob/6e8e1ed77a7da38a807f0bdb2bc2b70ae62c0c59/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L120

`deprecate` throws a `FutureWarning` which is better IMO compared to `logger.warning` here

Reply:
Oh cool! See if the latest [updates](https://github.com/huggingface/diffusers/pull/2918/commits/0a974969ebe2809c4668a6b50be2b97f74131617) are good?